agent/containers/images/Makefile

```diff
@@ -66,105 +98,92 @@ _WORKLOAD_RPMS = fio uperf
+_DISTROS := $(foreach d,${_ALL_DISTRO_NAMES},${_ALL_${d}_VERSIONS})
+
 # By default we only build images for the following distributions:
-_DISTROS = centos-8 centos-7 fedora-33 fedora-32
+_DEFAULT_DISTROS = centos-8 fedora-33 fedora-34
 
 # By default we won't build the Tool Data Sink and Tool Meister images
-all: all-tags $(foreach distro, ${_DISTROS}, ${distro}-all-tagged)
-
-tds: all-tags $(foreach distro, ${_DISTROS}, ${distro}-tool-data-sink-tagged)
+all: pkgmgr-clean all-tags $(_DEFAULT_DISTROS:%=%-all-tagged)
 
-tm: all-tags $(foreach distro, ${_DISTROS}, ${distro}-tool-meister-tagged)
+tds: pkgmgr-clean all-tags $(_DEFAULT_DISTROS:%=%-tool-data-sink-tagged)
 
-# We also offer targets per distribution target
-centos-8: all-tags centos-8-all-tagged
-
-centos-7: all-tags centos-7-all-tagged
-
-fedora-33: all-tags fedora-33-all-tagged
-
-fedora-32: all-tags fedora-32-all-tagged
+tm: pkgmgr-clean all-tags $(_DEFAULT_DISTROS:%=%-tool-meister-tagged)
 
-centos-8-tds: all-tags centos-8-tool-data-sink-tagged
-
-centos-7-tds: all-tags centos-7-tool-data-sink-tagged
-
-fedora-33-tds: all-tags fedora-33-tool-data-sink-tagged
+#+
+# For the following rule patterns, the "%" represents the "distribution" name,
+# e.g., as derived from the _DEFAULT_DISTROS list.
+#
+# The string matching the "%" is called the "stem", in GNU Make parlance.  The
+# "$*" references are replaced with that stem value.
+#
+# See https://www.gnu.org/software/make/manual/make.html#Automatic-Variables
+#-
 
-fedora-32-tds: all-tags fedora-32-tool-data-sink-tagged
+# We also offer targets per distribution target
+${_DISTROS}: %: pkgmgr-clean %-all-tagged
 
-centos-8-tm: all-tags centos-8-tool-meister-tagged
+$(_DISTROS:%=%-tds): %-tds: pkgmgr-clean %-tool-data-sink-tagged
 
-centos-7-tm: all-tags centos-7-tool-meister-tagged
+$(_DISTROS:%=%-tm): %-tm: pkgmgr-clean %-tool-meister-tagged
 
-fedora-33-tm: all-tags fedora-33-tool-meister-tagged
+# For any given target, extract the distribution name and version from the
+# first two fields, e.g., fedora-35-tds would yield values "fedora" and "35"
+# for DIST_NAME and DIST_VERSION, respectively.
+%: DIST_NAME = $(wordlist 1, 1, $(subst -, ,$*))
+%: DIST_VERSION = $(wordlist 2, 2, $(subst -, ,$*))
 
-fedora-32-tm: all-tags fedora-32-tool-meister-tagged
 
-#+
 # Tagging targets
-#-
-
-# Add the "latest" tag to the local images.
-tag-latest: $(foreach distro, ${_DISTROS}, ${distro}-tag-latest)
-
-# Add the "beta" tag to the local images.
-tag-beta: $(foreach distro, ${_DISTROS}, ${distro}-tag-beta)
-
-# Add the "alpha" tag to the local images.
-tag-alpha: $(foreach distro, ${_DISTROS}, ${distro}-tag-alpha)
-
-# Add the "v<Major>-latest" tag to the local images.
-tag-major: $(foreach distro, ${_DISTROS}, ${distro}-tag-major)
+#
+# Available tags values are "latest", "alpha", "beta", "major", and "major-minor"
+# "major" produces a tag of "v<Major>-latest", and "major-minor" produces a tag
```

Comment:
Does Major Minor hang out with General Specialist?

Reply:
😁

I didn't write that...I just moved it there.  😉

Reply:
Not a comment: just a random brain emission as I read the conjunction. 😆